### ISSUE TITLE
Use getAddonTypeFilter in a more direct way

### ISSUE THREAD

**Issue by rebmullin** [mozilla/addons-frontend#5888]
### Describe the problem and steps to reproduce it:

maybe this could be done in a more direct way (vs sprinkled throughout the code). For example in the api:

```javascript
if (filters.addonType) {
  filters.addonType = getAddonTypeFilter(addonType),
}
```

note: Once migration of lightweight happens this will all be removed so just something to think about too

**Comment by willdurand** [comment]
@kumar303 I wonder why this is not done at the component level actually? That's usually what we do when we need to pass filters from the UI to the API layer, avoiding to patch the saga layer (like #5885) and/or the API layer (as suggested here).

It is a nice hack but it is a hack we could avoid (not the kind of hack we need for faking static theme support on Android). I'd probably change the logic in the upper layers (like in the components) rather than adding such hacks. What did I miss?

**Comment by kumar303** [comment]
Yes, maybe it is more direct to change it at the component level, I'm not sure. I find the logic for handling filters confusing and hard to follow.

**Comment by rebmullin** [comment]
These page types - home, themes landing, etc -  are calling the search api to get their data so I am not sure how it could be done anywhere else (aside from directly in the api) in these instances. let me know if I am missing something..

I like the idea of adding to the api level b/c it seems like it might be a little bit more contained (and less confusing..?). Also, since it's somewhat temporary anyway, it might be easier to handle the [cleanup and switch over](#5359) after the migration.

**Comment by willdurand** [comment]
There is a lot of old code in the landing pages (valid addon types, mapping with visible addon types, etc.), and I'd like to refactor them (it? since it is a component) instead of patching the sagas or api.

There should not be a lot of logic in the sagas, it is the glue between redux and the api, via the action creators. The problem with patching the saga is that we are going to dispatch "fetch addonType = persona" (visible in redux devtools) and the request will be "/api/search?addonType=statictheme" (in the network panel) because the `addonType` is patched in the saga.

It's fine as a hotfix but the long term solution should be to update the `LandingPage` to support static themes, just like what has been done everywhere else IMO.

**Comment by willdurand** [comment]
Yeah, definitely!

**Comment by willdurand** [comment]
Closing this issue because we do not have to support LWT and ST. We only have ST so we can get rid of this function as stated by the OP:

> note: Once migration of lightweight happens this will all be removed so just something to think about too
